src/components/Todos.vue

```diff
@@ -22,13 +46,43 @@ import { mapGetters, mapActions } from 'vuex'
 export default {
     name: 'Todos',
     methods: {
-        ...mapActions(['fetchTodos', 'deleteTodo', 'fetchCurrentUser']),
+        ...mapActions(['fetchTodos', 'deleteTodo', 'fetchCurrentUser', 'completeTodo', 'selectTodo','updateTodo']),
         onDeleteClickHandler(id){            
             this.deleteTodo(id);
             
         },
+        onCheckBoxChecked(todo){          
+          this.completeTodo(todo);
+        },
+        onEditButtonClick(todo){
+          this.todoData.id = todo.id;
+          this.todoData.title = todo.title;
+          this.todoData.description = todo.description;
+          this.todoData.priority = todo.priority;
+          this.todoData.completed = todo.completed;
```

Comment:
`this.todoData = {...todo}`
ili ubaci lodash pa onda `_.cloneDeep(todo)`
ovo drugo je pametnije, ali okej ti je i ovo prvo